repos/system_upgrade/common/libraries/overlaygen.py

```diff
@@ -42,39 +171,74 @@ def _mount_name(mountpoint):
 def _mount_dir(mounts_dir, mountpoint):
     return os.path.join(mounts_dir, _mount_name(mountpoint))
 
+def _get_mountpoint(mount_points, dir_path):
+    for mp in sorted(mount_points, reverse=True):
+        mod_mp = mp if mp[-1] == '/' else '{}/'.format(mp)
+        if dir_path.startswith(mod_mp):
+            # longest first, so the first one we find, is the last mp on the path
+            return mp
```

Comment:
If the mountpoint semantics of this function are the same as for the `df` command, then this function returns incorrect mountpoint if called as `_get_mountpoint(['/var/lib', '/var'], '/var/lib')`

Reply:
Good catch! I will check it

Reply:
Ah. Now I've got the idea why i did it in this way. It was expected that dir_path will be always e.g. scratch_dir or something like that, which will have always mountpoint above. But I changed the name of the function and it's not clear from that now. So either the funcionality should be updated or the name/docstring to make it clear. Also `_get_mountpoint` / `_get_mountpoints` is a little bit not ideal as these functions are very close to each other. So I vote for the name change.

Reply:
@MichalHe  renamed to `_get_scratch_mountpoint`